tests/unit/amo/pages/TestGuides.js

```diff
@@ -68,9 +69,32 @@ describe(__filename, () => {
       dispatchSpy,
       fetchGuidesAddons({
         guids,
-        errorHandlerId: root.instance().props.errorHandler.id,
+        errorHandlerId: errorHandler.id,
+      }),
+    );
+
+    sinon.assert.calledOnce(dispatchSpy);
+  });
+
+  it('does not fetch guides addons while loading', () => {
+    const { store } = dispatchClientMetadata();
+    const slug = 'privacy';
+    const errorHandler = createStubErrorHandler();
+    const sections = getSections(slug, fakeI18n());
+    const guids = sections.map((section) => section.addonGuid);
+
+    store.dispatch(
+      fetchGuidesAddons({
+        guids,
+        errorHandlerId: errorHandler.id,
       }),
     );
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
   });
```

Comment:
We could have a similar test case to load empty results and make sure it does not dispatch (the second part of the `if` in the constructor).